frontend/src/features/map_planning/components/BaseLayerConfigurator.tsx

```diff
@@ -0,0 +1,197 @@
+import { Layer, Line } from "react-konva";
+import BaseLayer from "../layers/BaseLayer";
+import { BaseStage } from "./BaseStage";
+import { MAP_PIXELS_PER_METER } from "../utils/Constants";
+
+enum MeasurementState {
+    Initial,
+    OnePointSelected,
+    TwoPointsSelected,
+};
+
+const BaseLayerConfigurator = () => {
+    const [imageUrl, setImageUrl]      = useState('');
+    const [rotation, setImageRotation] = useState(0);
+    const [scale, setImageScale]       = useState(10);
+    const [realWorldLength, setRealWorldLength] = useState(0);
+    
+    const onUrlInputChange = (value: string | number) => {
+        // TODO: add error handling 
+        if (typeof value === 'number') return;
+        setImageUrl(value);
+    }  
+    
+    const onRotationInputChange = (value: string | number) => {
+        if (typeof value === 'string') return;
+        setImageRotation(value);
+    }
+    
+    const onScaleInputChange = (value: string | number) => {
+        if (typeof value === 'string') return;
+        setImageScale(value);
+    }
+
+    const onMetersInputChange = (value: string | number) => {
+        if (typeof value === 'string' || Number.isNaN(value)) return;
+        
+        const centimeters = realWorldLength - Math.floor(realWorldLength);
+        setRealWorldLength(value + centimeters);
+
+        console.log(realWorldLength);
+    }
+
+    const onCentimetersInputChange = (value: string | number) => {
+        if (typeof value === 'string' || Number.isNaN(value)) return;
+
+        const meters = Math.floor(realWorldLength);
+        setRealWorldLength(meters + (value / 100));
+        console.log(realWorldLength);
+    }
+
+    // Determine the scale of the image using the length of a known distance
+    const [measureState, setMeasureState]           = useState(MeasurementState.Initial);
+    const [measureLinePoints, setMeasureLinePoints] = useState<number[]>([]);
+    const [measuredLength, setMeasuredLength]       = useState(0);
+
+    const [showDistanceInputModal, setShowDistanceInputModal] = useState(false);
+
+    const onBaseStageClick = (e: KonvaEventObject<MouseEvent>) => {
+        if (e.evt.button !== 0) return;
+        
+        switch (measureState) {
+            case MeasurementState.Initial:
+                const x =  e.target.getStage()?.getRelativePointerPosition()?.x == null ? 0 : e.target.getStage()?.getRelativePointerPosition()?.x;
+                const y =  e.target.getStage()?.getRelativePointerPosition()?.y == null ? 0 : e.target.getStage()?.getRelativePointerPosition()?.y;
+                
+                setMeasureLinePoints([x ?? 0, y ?? 0]);
+                setMeasureState(MeasurementState.OnePointSelected);
+                break;
+            
+            case MeasurementState.OnePointSelected:
+                const lineLengthX = Math.abs(measureLinePoints[2] - measureLinePoints[0]);
+                const lineLengthY = Math.abs(measureLinePoints[3] - measureLinePoints[1]);
```

Comment:
curious: is it guaranteed that these indices will not be undefined or null?

Reply:
The configurator operates as a state machine with three possible states.
1. No point selected on the given image.
2. One Point selected.
3. Two Points selected.

The code you pointed out should not be executed unless unless the configurator is in the third state.
In theory this implies that all four components of the array are set (unless there is a bug of course).

Although I have to admit that some sort of check/assert and more documentation might be appropriate here.